### PR TITLE
improve doc for 'lineinfile'

### PR DESCRIPTION
**Pull request by computersalat**
##### SUMMARY
improve doc for *lineinfile* especially for 'regexp:' and 'insertafter:'

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
 lib/ansible/modules/files/lineinfile.py

##### ANSIBLE VERSION
```
ansible 2.2.1.0
```

### CODE REVIEW

File: lib/ansible/modules/files/lineinfile.py [hunk 1]
```diff
@@ -53,10 +53,10 @@
     version_added: 1.7
     description:
       - The regular expression to look for in every line of the file. For
-        C(state=present), the pattern to replace if found; only the last line
-        found will be replaced. For C(state=absent), the pattern of the line
-        to remove.  Uses Python regular expressions; see
-        U(http://docs.python.org/2/library/re.html).
+        C(state=present), the pattern to replace if found (even if you use insertafter
+        or insertbefore); only the last line found will be replaced.
```
Comment (dharmabumstead) [inline lib/ansible/modules/files/lineinfile.py]: This wording is confusing. Did you mean to say "even if you use insertbefore or insertafter, only the last line found will be replaced"? If so, please remove the parenthesis. Thanks.

**Review by dharmabumstead — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/files/lineinfile.py (hunk 1 above)
Comment (computersalat) [reply lib/ansible/modules/files/lineinfile.py]: If I would meant it this way I would have written it this way ... perhaps ``` even ``` should be replaced with ``` also ```.
Sorry I am not a native English speaker ...

**Comment by computersalat** [comment]
all-in-all it is somehow awesome how fast this issue is handled  ...

**Review by dharmabumstead — APPROVED** [review]

**Comment by acozine** [comment]
Hi @computersalat - thanks for contributing back to Ansible via this PR. My apologies for the long wait and thank you for your patience! 

I tested the `lineinfile` module today. You're right, the current documentation doesn't describe the interaction of the  `insertbefore` and `insertafter` parameters with the `regexp` parameter. But your suggested changes to the docs don't quite capture the behavior either. The `regexp` parameter takes priority. For example, if you start with this file: 

```
First line
Second line
Third line
```

and run this task against it:

```
    - name: finds match to regexp, replaces it in current location, ignores insertafter
      lineinfile:
        path: /path/to/file
        state: present
        insertafter: ^Second
        regexp: ^First
        line: "First line replaced"
```

you'll get:

```
First line replaced
Second line
Third line
```

if you run this task instead:

```
    - name: finds no match to regexp, finds match to insertafter, adds line after insertafter match
      lineinfile:
        path: /path/to/file
        state: present
        insertafter: ^Second
        regexp: ^New
        line: "New line after Second line"
```

you'll get:

```
First line
Second line
New line after Second line
Third line
```

I've opened a new PR to update the module docs. Take a look at #44277 and let me know if it addresses your concerns. Thanks!

**Comment by computersalat** [comment]
@acozine 
sorry, but this is too long ago to remember how this `lineinfile` is working ...
I just tried to improve the docs in the way I found out how it is working ...
... currently I have no time to check this again, sorry
You seem to know this module better than me ... please update the docs (examples) in a way that a noob (like me) is able to understand how it works ... with life examples and meaningful comments.
Thank you

**Comment by samdoran** [comment]
Superseded by #44277

**Pull request closed without merging**